llvm/lib/Target/AMDGPU/Disassembler/AMDGPUDisassembler.cpp

```diff
@@ -38,9 +38,16 @@ using namespace llvm;
 
 #define DEBUG_TYPE "amdgpu-disassembler"
 
-#define SGPR_MAX                                                               \
-  (isGFX10Plus() ? AMDGPU::EncValues::SGPR_MAX_GFX10                           \
-                 : AMDGPU::EncValues::SGPR_MAX_SI)
+enum : unsigned {
```

Comment:
It seems the encoding side isn't interested in the range boundaries much (which makes sense?). `LITERAL_CONST` could be used there, but currently isn't, so maybe it can wait for when we need it in a wider scope? (I wouldn't mind it either way.)

Reply:
I was specifically surprised that LITERAL_CONST isn't used. I do not see moving it to decoding side a good thing, I would rather replace uses of 255 on the encoding side.

Reply:
OK. What namespace do we want `LITERAL_CONST` to be declared in? Would it make sense to rename `EncValues`  to `HWEncoding` and introduce another `EncValues` namespace for that kind of constants?

Reply:
I do not have strong preference here, but why don't we start with replacing uses of literals with symbolic names in the encoder and see what are we really using?